token/program2/src/state.rs

```diff
@@ -36,16 +38,42 @@ pub struct Account {
     /// If `delegate` is `Some` then `delegated_amount` represents
     /// the amount authorized by the delegate
     pub delegate: COption<Pubkey>,
-    /// Is `true` if this structure has been initialized
-    pub is_initialized: bool,
+    /// Whether the account has been initialized or frozen
+    pub is_initialized: AccountState,
     /// Is this a native token
     pub is_native: bool,
     /// The amount delegated
     pub delegated_amount: u64,
 }
+impl Account {
+    /// Checks if account is frozen
+    pub fn is_frozen(&self) -> bool {
+        self.is_initialized == AccountState::Frozen
+    }
+}
 impl IsInitialized for Account {
```

Comment:
Do we need this trait any longer?  Could be a member function of `Account` like `is_frozen`?

Reply:
It's still used for Mint and Multisig unpack, so I'm leaving for now. We can consider removing in a separate pr.